app/src/stores/app.ts

```diff
@@ -13,4 +16,9 @@ export const useAppStore = defineStore({
 		accessTokenExpiry: 0,
 		basemap: 'OpenStreetMap',
 	}),
+	actions: {
+		setSidebarOpen(isOpen: boolean) {
+			this.sidebarOpen = sidebarOpenObject.setValue(isOpen);
+		},
+	},
```

Comment:
This might be a potential footgun as `sidebarOpen` would still be modifiable through `useAppStore().sidebarOpen`.

Reply:
How to avoid it?

Reply:
One option would be to convert `appStore` to a [setup store](https://pinia.vuejs.org/cookbook/composables.html#setup-stores) as mentioned above.

Reply:
Thanks for the tip 🙌 I've made some changes to avoid this problem.